src/resources/api_nw_newwin.js

```diff
@@ -760,7 +760,11 @@ apiBridge.registerCustomHook(function(bindingsAPI) {
       callback(cwindows.map(create_nw_win));
     });
   });
-
+  apiFunctions.setHandleRequest('isDevToolsOpen', function() {
+    return chrome.windows.getAll({ populate: true, windowTypes:['devtools'] }, function( wins ) {
+      return wins.length > 0;
+    })
+  });
```

Comment:
```suggestion
  apiFunctions.setHandleRequest('isDevToolsOpen', function(callback) {
    return chrome.windows.getAll({ populate: true, windowTypes: ['devtools'] }, function(wins) {
      callback(wins.length > 0);
    })
  });
```

Reply:
I think this should fix the async issue. Example usage (assuming this is how the API was implemented pre v13):

```js
nw.Window.isDevToolsOpen((status) => {
  console.log(status)
});
```

Reply:
Tested locally and seems to work nicely:

![Screenshot (16)](https://github.com/user-attachments/assets/f2a78479-8912-4122-9dca-1dcd977dbcb2)

Reply:
@rogerwang what are your thoughts?

Reply:
Your fix is good. Thanks.

Reply:
done, thank for your time friend

Reply:
@rogerwang does a similar change need to be made on src/resources/nw_api_window.js?